### PR TITLE
Toggle defeated state for npcs when healed to above 0 hp when defeated.

### LINKED ISSUE #3806: mark as defeated not removed when healed
to reproduce: 
1. put an NPC in combat
2. roll a damage roll
3. apply damage until marked as defeated
4. use the heal button on the damage roll

the Defeated marker does not Disappear
Removing defeated on the combat tracker does not remove the skull overlay
removing the skull overlay removes the defeated state from the combat tracker

### PR DESCRIPTION
**Pull request by xdy**
Fixes #3806

### CODE REVIEW

**Review by stwlam — CHANGES_REQUESTED** [review]

File: src/module/actor/base.ts [hunk 1]
```diff
@@ -596,6 +596,7 @@ class ActorPF2e extends Actor<TokenDocumentPF2e, ItemTypeMap> {
 
     /* -------------------------------------------- */
     /*  Rolls                                       */
+
```
[the remark this answers is not in the document]
Comment (xdy) [reply src/module/actor/base.ts]: Huh. I must have accidentally run my ide:s formatter after lint:fix (not that I understand why it'd want *that* line added.) Anyhoo, removed all the spurious formatting and reran lint:fix

File: src/module/actor/base.ts [hunk 2]
```diff
@@ -1272,7 +1275,6 @@ interface ActorPF2e extends Actor<TokenDocumentPF2e, ItemTypeMap> {
         data: PreCreate<foundry.data.ActiveEffectSource>[] | PreCreate<ItemSourcePF2e>[],
         context?: DocumentModificationContext
     ): Promise<ActiveEffectPF2e[] | ItemPF2e[]>;
-
```
Comment (stwlam) [inline src/module/actor/base.ts:1275]: This one too

Comment (xdy) [reply src/module/actor/base.ts]: Guess I got overzealous with removing the inserted lines. Pushed update.

File: src/module/actor/base.ts [hunk 3]
```diff
@@ -755,10 +755,9 @@ class ActorPF2e extends Actor<TokenDocumentPF2e, ItemTypeMap> {
...
             await this.update(hpUpdate.updates);
         }
-        if (this.hitPoints?.value === 0) {
-            const deadAtZero = game.settings.get("pf2e", "automation.actorsDeadAtZero");
-            if (this.type === "npc" && ["npcsOnly", "both"].includes(deadAtZero) && !token.combatant?.isDefeated) {
-                token.combatant?.toggleDefeated();
+        const deadAtZero = ["npcsOnly", "both"].includes(game.settings.get("pf2e", "automation.actorsDeadAtZero"));
+        if (this.isOfType("npc") && deadAtZero && this.isDead !== token.combatant?.isDefeated) {
+            token.combatant?.toggleDefeated();
             }
```
Comment (stwlam) [inline src/module/actor/base.ts:761]:
```suggestion
```

**Review by stwlam — APPROVED** [review]

**Pull request merged**